File: app/src/main/java/com/coinninja/coinkeeper/ui/memo/MemoCreator.java
package com.coinninja.coinkeeper.ui.memo;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.coinninja.android.helpers.Input;
import com.coinninja.coinkeeper.R;
import com.coinninja.coinkeeper.view.dialog.GenericAlertDialog;

import javax.inject.Inject;

public class MemoCreator {
    private OnMemoCreatedCallback callback;
    private GenericAlertDialog genericAlertDialog;
    private View view;

    @Inject
    MemoCreator() {
    }

    public void createMemo(@NonNull AppCompatActivity activity, @NonNull OnMemoCreatedCallback callback, String text) {
        this.callback = callback;
        view = LayoutInflater.from(activity).inflate(R.layout.dialog_create_memo, null);
        genericAlertDialog = GenericAlertDialog.newInstance(view, false, false);
        genericAlertDialog.asWide();
        genericAlertDialog.show(activity.getSupportFragmentManager(), MemoCreator.class.getSimpleName());
        EditText memoView = view.findViewById(R.id.memo);
        memoView.setText(text);
        memoView.setSelection(text.length());
        view.findViewById(R.id.done).setOnClickListener(v -> onDonePressed());
        memoView.postDelayed(() -> {
            Input.INSTANCE.showKeyboard(memoView);
        }, 200);

    }

    private void onDonePressed() {
        TextView memo = view.findViewById(R.id.memo);
        String text = memo.getText().toString().trim();

        if (!"".equals(text))
            callback.onMemoCreated(text);

        genericAlertDialog.dismiss();
        genericAlertDialog = null;
        view = null;
        callback = null;
    }


    public interface OnMemoCreatedCallback {
        void onMemoCreated(String memo);
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/ui/memo/MemoCreator.kt
package com.coinninja.coinkeeper.ui.memo

import android.view.LayoutInflater
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import app.dropbit.annotations.Mockable
import app.dropbit.commons.util.isNotNullOrEmpty
import com.coinninja.android.helpers.Input.showKeyboard
import com.coinninja.coinkeeper.R.id
import com.coinninja.coinkeeper.R.layout
import com.coinninja.coinkeeper.view.dialog.GenericAlertDialog
import javax.inject.Inject

@Mockable
class MemoCreator @Inject internal constructor() {

    @JvmOverloads
    fun createMemo(activity: AppCompatActivity, callback: OnMemoCreatedCallback, text: String = "") {
        val view = LayoutInflater.from(activity).inflate(layout.dialog_create_memo, null)
        GenericAlertDialog.newInstance(view, false, false).also { genericAlertDialog ->
            genericAlertDialog.asWide()
            view.also {
                val memoView: EditText = it.findViewById(id.memo)
                memoView.setText(text)
                memoView.setSelection(text.length)
                it.findViewById<View>(id.done).setOnClickListener {
                    val text = memoView.text.toString().trim { it <= ' ' }
                    if (text.isNotNullOrEmpty()) callback.onMemoCreated(text)
                    genericAlertDialog.dismiss()
                }
                it.postDelayed({ showKeyboard(memoView) }, 200)
            }
        }.show(activity.supportFragmentManager, MemoCreator::class.java.simpleName)

    }

    interface OnMemoCreatedCallback {
        fun onMemoCreated(memo: String)
    }
}